src/status_im2/contexts/syncing/scan_sync_code/view.cljs

```diff
@@ -255,136 +252,102 @@
-                                       connection-string]
-                                      300)
+      (do
+        (set-qr-code-succeeded)
+        (debounce/debounce-and-dispatch [:syncing/input-connection-string-for-bootstrapping
+                                         connection-string]
+                                        300))
       (rf/dispatch [:toasts/upsert
                     {:icon           :i/info
                      :icon-color     colors/danger-50
                      :override-theme :light
                      :text           (i18n/label :t/error-this-is-not-a-sync-qr-code)}]))))
 
-(defn render-camera
-  [show-camera? torch-mode qr-view-finder camera-ref on-read-code]
-  (when (and show-camera? (:x qr-view-finder))
-    [:<>
-     [rn/view {:style style/camera-container}
-      [camera-kit/camera
-       {:ref          #(reset! camera-ref %)
-        :style        style/camera-style
-        :camera-type  camera-kit/camera-type-back
-        :zoom-mode    :off
-        :torch-mode   torch-mode
-        :scan-barcode true
-        :on-read-code on-read-code}]]
-     [hole-view/hole-view
-      {:style style/hole
-       :holes [(merge qr-view-finder
-                      {:borderRadius 16})]}
-      [blur/view
-       {:style            style/absolute-fill
-        :blur-amount      10
-        :blur-type        :transparent
-        :overlay-color    colors/neutral-80-opa-80
-        :background-color colors/neutral-80-opa-80}]]]))
+(defn- render-camera
+  [{:keys [torch-mode qr-view-finder set-qr-code-succeeded]}]
+  [:<>
+   [rn/view {:style style/camera-container}
+    [camera-kit/camera
+     {:style        style/camera-style
+      :camera-type  camera-kit/camera-type-back
+      :zoom-mode    :off
+      :torch-mode   torch-mode
+      :scan-barcode true
+      :on-read-code #(check-qr-code-and-navigate % set-qr-code-succeeded)}]]
+   [hole-view/hole-view
+    {:style style/hole
+     :holes [(merge qr-view-finder {:borderRadius 16})]}
+    [blur/view
+     {:style            style/absolute-fill
+      :blur-amount      10
+      :blur-type        :transparent
+      :overlay-color    colors/neutral-80-opa-80
+      :background-color colors/neutral-80-opa-80}]]])
 
-(defn f-view
-  [{:keys [title show-bottom-view? background animated?]}]
-  (let [insets             (safe-area/get-insets)
-        active-tab         (reagent/atom 1)
-        qr-view-finder     (reagent/atom {})
-        render-camera?     (reagent/atom false)
-        torch?             (reagent/atom false)
-        app-state-listener (atom nil)]
+(defn- reset-animations-and-navigate-fn
+  [{:keys [render-camera? show-camera?] :as params}]
+  (letfn [(reset-fn []
+            (rf/dispatch [:navigate-back])
+            (when @dismiss-animations (@dismiss-animations))
+            (animation/reset-animations params))]
     (fn []
-      (let [camera-ref (atom nil)
-            read-qr-once? (atom false)
```

Comment:
These atoms shouldn't be declared in the render function since they will be recreated in each render

Reply:
@ulisesmac - The `on-read-code` function from the `react-native-camera-kit` library is called as long as a QR code is in the frame of the camera (or the user is focusing on it). To prevent this, we added the `read-qr-once?` atom within the render and reset it after 3 seconds on a successful read.

This also allows the user to scan any QR code after 3 seconds if they accidentally scanned the wrong one. 

The present code will result in endless error toasts if the user scans a wrong PR.

Reply:
Hmm, is there any way to add a unit test (component spec fine) to ensure this feature is retained?

Reply:
@smohamedjavid 
The present code is using a different approach without using the atom in the render function and also fixing the bug for this issue.

What would be a wrong QR?
Do you know if that endless toast behavior is reproducible in this branch? Could you please tell me how to reproduce it?

Thank you so much!

Reply:
@ulisesmac 

> What would be a wrong QR?

A QR code which is NOT a sync QR (QR data would start with `cv2:`). 

> Do you know if that endless toast behavior is reproducible in this branch? Could you please tell me how to reproduce it?

Yes, Kindly download and install this PR build (https://github.com/status-im/status-mobile/pull/16746#issuecomment-1646374593) into your device. Point your device at any QR code from the internet.

It doesn't need to be a wrong QR. The camera library fires the `on-read-code` method (with QR data) as long as a QR is in focus. This will happen with the correct sync QR too (initiates the local pairing repeatedly by calling the native module as long as a Correct Sync QR is in focus).

<hr />

I fixed a similar issue https://github.com/status-im/status-mobile/issues/15970 (PR: https://github.com/status-im/status-mobile/pull/16012) with the help of `view-id` in `app-db`.

Reply:
> Hmm, is there any way to add a unit test (component spec fine) to ensure this feature is retained?

I don't know as this is a ratom inside render (not a prop). But, I will see if we can add a check. 

P.S. I would add a comment explaining why we need it.

Reply:
@smohamedjavid 

The problem I see with a non-reactive atom within the render function is that we are not able to know what's the value stored by the atom: Let's say we reset the atom, then we `deref` it, if a re-render happens just before, our value will be missed.

Thanks for noticing this case, I didn't test with a wrong QR (so bad).

I added a reagent  control atom to avoid multiple calls to `on-read` (not inside the render function), also the toast call is using `debounce-and-dispatch` in the same way that when we are navigating. :+1: 

The commit where I added is 2279613f7f249bc90eecc3c8c8d90e00bb829358,
please review again and approve if this is ok for you, otherwise we can check what else is missing :)

Thanks!